### ISSUE TITLE
Upgrade question Syno Docker controller

### ISSUE THREAD

**Issue by gjs1969** [jacobalberty/unifi-docker#81]
### Host operating system

Synology DiskStation / Docker controller

I still have 5.5.24 version running and want to upgrade to the 5.6 version. Normally i would use the following command:

docker run -d -p 3478:3478/udp -p 8080:8080 -p 8443:8443 -v /volume1/docker/unifi/:/var/lib/unifi --restart=always --name unifi jacobalberty/unifi:latest

Now with the new folder structure, can i still use it in this way ? Or do i have to do preparations before i can upgrade ? It it said that one should switch to new folder layout asap, but i do not know how. Simply replace folder names in above command would not be right i guess, because the folders (unifi/data - unifi/log) do not exist at the moment. 

So basically, how to upgrade from 5.5.24 to 5.6.xx ? Many thanx,

**Comment by jacobalberty** [comment]
You can just pull the newest image and it will upgrade fine, the new volume layout is a separate issue entirely.

The new volume layout should be entirely backwards compatible with the old, it just provides a cleaner way of organizing the data to allow you to put data and log in a single volume.

You have a couple of options if you want to move to the new layout, the easiest is to only move your data  volume line. The new docker run command would look like
`docker run -d -p 3478:3478/udp -p 8080:8080 -p 8443:8443 -v /volume1/docker/unifi/:/unifi/data --restart=always --name unifi jacobalberty/unifi:latest` 
The other option is to move the contents of /volume1/docker/unifi as a whole into a data subdirectory. Something like
```
mkdir -p /volume1/docker/unifi2/
mv /volume1/docker/unifi /volume1/docker/unifi2/data
mv /volume1/docker/unifi2 /volume1/docker/unifi
```
then your docker run command would be
`docker run -d -p 3478:3478/udp -p 8080:8080 -p 8443:8443 -v /volume1/docker/unifi/:/unifi --restart=always --name unifi jacobalberty/unifi:latest`

This would give you the advantage of having your log files and data folder preserved under /volume1/docker/unifi. 

Of course to just upgrade from 5.5.24 to 5.6.xx you don't have to do any of this, the new version of the image is backwards compatible with the old volume layout.

**Comment by gjs1969** [comment]
Well, that is a very complete answer, thank you so much ! I think i will try the new layout on a test environment. Thank you again for your time and information 🥇
